tests/e2e/utils/requestHandlers/CheApiRequestHandler.ts

```diff
@@ -0,0 +1,43 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import axios, { AxiosResponse } from 'axios';
+import { TestConstants } from '../../TestConstants';
+import { TYPES } from '../../inversify.types';
+import { inject, injectable } from 'inversify';
+import { IHeaderHandler } from './IHeaderHandler';
+
+@injectable()
+ export class CheApiRequestHandler {
+
+    constructor(@inject(TYPES.HeaderHandler) private readonly headerHandler: IHeaderHandler) {
+    }
+
+    async get(url: string) : Promise<AxiosResponse> {
+        return await axios.get(this.assembleUrl(url), await this.headerHandler.getHeaders());
+    }
+
+    async post(url: string, data?: string) : Promise<AxiosResponse> {
+        if ( data === undefined ) {
+            return await axios.post(this.assembleUrl(url), await this.headerHandler.getHeaders());
+        } else {
```

Comment:
Redundant ```else```
```
    async post(url: string, data?: string): Promise<AxiosResponse> {
        if (data === undefined) {
            return await axios.post(this.assembleUrl(url), await this.headerHandler.getHeaders());
        }

        return await axios.post(this.assembleUrl(url), data, await this.headerHandler.getHeaders());
    }
```

Reply:
I don't see much difference in this specific case. I understand that the form you are suggesting could be very beneficial in some other cases.
If it would be for my personal feeling, then *here* (and I emphasize *here*), the version with `else` is a bit more nicer to my eyes.

Reply:
Here you can read an extended description of the rule I mentioned.
Of course, this is not a very serious resource, but if you want
I can find a more serious source.
https://www.geeksforgeeks.org/writing-clean-else-statements

Reply:
I would suggest to prefer `If` without `else` where it's possible, because it shorten possible execution paths and so makes the code easier to read and understand. 
This specific case Smalltalk calls "guard clauses" https://softwareengineering.stackexchange.com/a/18459

Reply:
Heh... This conversation about ifs and elses is starting to be almost "decadent" :-D
Nobody is questioning, that the approach you are suggesting here and all the resources you are linking here are talking about is the right approach in general and in most of the cases, but in **some specific cases** writing `else` statement could be "neutral" (nor good, nor bad) :-D
Anyway... To close this pointless discussion... Isn't this `if` unecessary after all? Because `axios.post` has `data` as optional:
```
post<T = any>(url: string, data?: any, config?: AxiosRequestConfig): AxiosPromise<T>;
```
Thus we don't have to care about that at all, isn't that right?

Reply:
Of course, you are right, if axios can apply null data. But I am not 100% that axios handle null data in the way we expected. Should be verified I guess.

Reply:
Ok, I'll try that and if it handles null data correctly, I'll push this.